File: platform/remote-driver/test-sdk/src/com/intellij/driver/sdk/VirtualFile.java
package com.intellij.driver.sdk;

import com.intellij.driver.client.Remote;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Remote("com.intellij.openapi.vfs.VirtualFile")
public interface VirtualFile {
  @NotNull String getName();

  @Nullable VirtualFile findChild(@NotNull String name);

  @Nullable VirtualFile findFileByRelativePath(@NotNull String relPath);
}

File: platform/remote-driver/test-sdk/src/com/intellij/driver/sdk/VirtualFile.kt
package com.intellij.driver.sdk

import com.intellij.driver.client.Remote

@Remote("com.intellij.openapi.vfs.VirtualFile")
interface VirtualFile {
  fun getName(): String
  fun getPath(): String

  fun findChild(name: String): VirtualFile?
  fun findFileByRelativePath(relPath: String): VirtualFile?

  fun delete(requestor: Any?)
  fun createChildData(requestor: Any, name: String): VirtualFile
}
